flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -539,51 +541,62 @@ private void startTriggeringCheckpoint(CheckpointTriggerRequest request) {
 									coordinatorsToCheckpoint, pendingCheckpoint, timer),
 							timer);
 
-			FutureUtils.assertNoException(
-				CompletableFuture.allOf(masterStatesComplete, coordinatorCheckpointsComplete)
-					.handleAsync(
-						(ignored, throwable) -> {
-							final PendingCheckpoint checkpoint =
-								FutureUtils.getWithoutException(pendingCheckpointCompletableFuture);
-
-							Preconditions.checkState(
-								checkpoint != null || throwable != null,
-								"Either the pending checkpoint needs to be created or an error must have been occurred.");
-
-							if (throwable != null) {
-								// the initialization might not be finished yet
-								if (checkpoint == null) {
-									onTriggerFailure(request, throwable);
-								} else {
-									onTriggerFailure(checkpoint, throwable);
-								}
+			CompletableFuture
```

Comment:
I think we could add `FutureUtils.assertNoException` here again if we used `handle` or `exceptionally` instead of `whenComplete` for the final exception handler.